__tests__/authRoutes.test.js

```diff
@@ -45,6 +47,69 @@ describe('User signup tests', () => {
   });
 });
 
+describe('verifying an email', () => {
+  let verificationCode;
+  before((done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send(userToVerify)
+      .end((err, res) => {
+        verificationCode = res.body.user.verificationToken;
+        done();
+      });
+  });
+  it('a successful email verification should return a success response object', (done) => {
+    chai.request(app)
+      .patch(`${url}/auth/verify?email=${userToVerify.email}&token=${verificationCode}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(200);
+        expect(res.body.message).to.eql('Email Verified');
+        expect(res.body.isVerified).to.eql(true);
+        done();
+      });
+  });
+  let secondVerificationCode;
+  before((done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send(secondUserToVerify)
+      .end((err, res) => {
+        secondVerificationCode = res.body.user.verificationToken;
+        done();
+      });
+  });
+  it('attempting to verify an email with invalid details should throw an error', (done) => {
+    chai.request(app)
+      .patch(`${url}/auth/verify?email=${secondUserToVerify.email}&token=${secondVerificationCode.slice(0, 34)}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(400);
+        expect(res.body.message).to.eql('Incorrect Credentials');
+        expect(res.body.isVerified).to.eql(false);
+        done();
+      });
+  });
+  let thirdVerificationCode;
+  before((done) => {
+    chai.request(app)
+      .post('/api/v1/auth/signup')
+      .send(thirdUserToVerify)
+      .end((err, res) => {
+        thirdVerificationCode = res.body.user.verificationToken;
+        done();
+      });
+  });
+  it('it should throw an internal server error upon email verification', (done) => {
+    const stub = sinon.stub(User, 'update');
+    const error = new Error('Something went wrong');
+    stub.yields(error);
+    chai.request(app)
+      .patch(`${url}/auth/verify?email=${thirdUserToVerify.email}&token=${thirdVerificationCode}`)
+      .end((err, res) => {
+        expect(res.status).to.eql(500);
+        done();
+      });
+  });
+});
```

Comment:
Review the statements in your `describe` and `it` blocks. They should always form a coherent sentence that explains what each test case does.